File: DP/climbing-stairs.py
# DP Bottom Up

class Solution:
    def climbStairs(self, n: int) -> int:
        # it is like a fibonnaci series
        one = 1
        two = 1
        if(n < 2):
            return 1
        else:
            for count in range(n - 1):
                temp = one
                one = one + two
                two = temp
        return one


# should print 8
print(Solution().climbStairs(5))

File: DP/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int left = 1;
        int right = 1;
        
        for(int i = 0; i < n - 1; i++) {
            int newLeft = left + right;
            right = left;
            left = newLeft;
        }
        
        return left;
    }
}